helix-view/src/editor.rs

```diff
@@ -165,6 +165,8 @@ pub struct Config {
     pub indent_guides: IndentGuidesConfig,
     /// Whether to color modes with different colors. Defaults to `false`.
     pub color_modes: bool,
+    /// File type icon/string map for overriding file type.
+    pub file_type_indicators: FileTypeConfig,
```

Comment:
if `FileTypeConfig` isn't doing any validation while deserializing, why not just use the HashMap directly?

Reply:
I could do that but this seems a bit more flexible. Does it come with any performance impact?

Reply:
I don't think it there would be a noticeable performance change. It's more about simplicity - using a custom struct adds a bunch of extra code for (de)serializing and deref that doesn't seem to add any value. If anything, you might be able to use a [type alias](https://doc.rust-lang.org/reference/items/type-aliases.html) to "rename" `HashMap<String, String>` but IMO it's still preferrable to just use the hashmap